pkg/handlers/wsserver.go

```diff
@@ -0,0 +1,110 @@
+package handlers
+
+import (
+	"errors"
+	"github.com/gorilla/websocket"
+	"github.com/kptm-tools/core-service/pkg/config"
+	"github.com/kptm-tools/core-service/pkg/interfaces"
+	"net/http"
+	"sync"
+)
+
+type WsServer struct {
+	cfg *config.Config
+	sync.RWMutex
+	// handlers are functions that are used to handle Events
+	handlers    map[string]EventHandler
+	clients     WsClientList
+	scanService interfaces.IScanService
+}
+
+var (
+	ErrEventNotSupported = errors.New("this event type is not supported")
+)
+
+var (
+	/**
+	websocketUpgrader is used to upgrade incomming HTTP requests into a persitent websocket connection
+	*/
+	websocketUpgrader = websocket.Upgrader{
+		// Apply the Origin Checker
+		CheckOrigin:     checkOrigin,
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+	}
+)
+
+// checkOrigin will check origin and return true if its allowed
+func checkOrigin(r *http.Request) bool {
+	return true
+}
+
+var _ interfaces.IWsHandlers = (*WsServer)(nil)
+
+func NewWsHandlers(scanService interfaces.IScanService) *WsServer {
+	server := &WsServer{
+		cfg:         config.LoadConfig(),
+		clients:     make(WsClientList),
+		handlers:    make(map[string]EventHandler),
+		scanService: scanService,
+	}
+	server.setupEventHandlers()
+	return server
+}
+func (m *WsServer) setupEventHandlers() {
```

Comment:
Ideally the receiver name on the same type `WsServer` throughout this file should always be the same to improve readability:

```suggestion
func (ws *WsServer) setupEventHandlers() {
```